codebase/superdarn/src.lib/tk/rpos.1.7/src/rpos_v2.c

```diff
@@ -0,0 +1,255 @@
+/* rpos_v2.c
+   =========
+   Author: E.G.Thomas
+
+   Copyright (C) 2022  Evan G. Thomas
+
+This file is part of the Radar Software Toolkit (RST).
+
+RST is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+Modifications:
+
+*/
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "radar.h"
+#include "rpos.h"
+#include "rpos_v2.h"
+#include "geodtgc.h"
+#include "rmath.h"
+
+
+/**
+ * This function converts a gate/beam coordinate to geographic
+ * position. The height of the transformation is given by height -
+ * if this value is less than 90 then it is assumed to be the
+ * elevation angle from the radar. If center is not equal to zero, then
+ * the calculation is assumed to be for the center of the cell, not the
+ * edge. The calculated values are returned in geocentric spherical
+ * coordinates (rho,lat,long). If model is equal to one the Chisham
+ * et al [2008] virtual height model is used; if model is equal to
+ * two the Thomas and Shepherd [2022] virtual height model is used;
+ * otherwise,the standard virtual height model is used.
```

Comment:
I recommend reformatting this as a list, so it is easier to read when scanning what the different `model` values mean.